### PR TITLE
test: add tests for GroupMemberSelector's member limit enforcement and callback behavior

### LINKED ISSUE #2911: Improve test code coverage for lib/views/after_auth_screens/chat/widgets/group_member_selector.dart
We will need unittests done for all methods, classes and functions found in this file.

Any widgets referenced in this file must also have unittests done

### PR Acceptance Criteria

1. When complete this file must show **100%** coverage when merged into the code base.
1. [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa/tree/develop/?displayType=list)
1. CodeRabbit must approve all your changes to the test file(s) and the file(s) under test
1. The PR will show a report for the code coverage for the file you have added. You can use that as a guide
1. No functionality must be removed to achieve this goal
1. All tests must be valid and truly test the code base

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by shshaheen**
### What kind of change does this PR introduce?

test: Adds widget tests for group member selection limits and callback behavior.

### Issue Number:

Fixes #2911 
### Did you add tests for your changes?

Yes

- [✓]  Tests are written for all changes made in this PR.
- [✓ ]  Test coverage meets or exceeds the current coverage (~90/95%).

Snapshots/Videos:
image

image

### If relevant, did you update the documentation?

Not relevant.

### Summary

This PR introduces widget tests for the GroupMemberSelector to validate critical member selection behavior.
The tests ensure that member selection is prevented when the group reaches the maximum allowed limit and that the onMembersChanged callback returns the correct set of selected members while excluding the current user.
With these additions, test coverage for this component has reached 100%, improving overall reliability and helping prevent regressions related to group size constraints.

### Does this PR introduce a breaking change?

No

### Checklist for Repository Standards

[✓]   Have you reviewed and implemented all applicable coderaabbitai review suggestions?
[✓]   Have you ensured that the PR aligns with the repository’s contribution guidelines?

### Other information

None.

### Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?

Yes

## Summary by CodeRabbit

* **Tests**
  * Renamed and reorganized widget tests for clarity.
  * Added helper utilities and constants to support stateful, callback-driven scenarios.
  * Expanded coverage for loading, empty, error, initialization and selection/deselection flows.
  * Added extensive edge-case coverage: null/missing data, large lists/virtualization, UI labels/subtitles, name/email/avatar displays, current-user behavior (top/non-selectable), aggregated selection text, and limit scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@coderabbitai perform a comprehensive review with the following checks:

## Testing & Coverage
1. Analyze all test files and verify they:
   - Cover all use cases including edge cases and error scenarios
   - Include valid assertions and meaningful test descriptions
   - Achieve 100% code coverage for new/modified code
   - Test both success and failure paths
   - Include integration tests where applicable
2. Identify specific lines/functions in implementation files that lack test coverage
3. Check for missing tests for:
   - Error handling and validation logic
   - Conditional branches and loops
   - Async operations and callbacks
   - State changes and side effects

## Code Quality
1. Review for:
   - Code duplication or violations of DRY principle
   - Proper error handling and logging
   - Security vulnerabilities (e.g., injection, XSS, authentication issues)
   - Performance concerns (inefficient algorithms, memory leaks)
   - Proper use of TypeScript types and avoiding `any`
   - Accessibility issues in UI components
2. Verify adherence to:
   - Repository coding standards and conventions
   - Consistent naming patterns
   - Proper component/function organization

## Documentation & Files
1. Validate that:
   - New functions/components have proper JSDoc/TSDoc comments
   - Complex logic includes inline explanatory comments
   - README or relevant docs are updated if needed
2. Identify unnecessary files:
   - Debug files, build artifacts, or temporary files
   - Unused imports or dead code
   - Files outside the scope of changes

**Implementation Correctness:** 
1. For the specific implementation file(s) being tested:
   - Verify the implementation logic is correct and free of bugs
   - Validate that queries target the correct database tables/entities
   - Check that foreign key relationships are correctly enforced
   - Confirm business logic constraints are properly implemented
   - Any bugs, logic errors, or incorrect implementations discovered in the file(s) under test MUST be fixed in this PR before approval

## Requirements & Compliance
1. Confirm the PR:
   - Fully addresses all requirements in the linked issue
   - Includes all acceptance criteria from the issue
   - Complies with all previously requested changes and resolved comments
2. Check for breaking changes and assess their necessity
3. Verify backward compatibility where applicable

## Review Decision
    1. When critical issues are found, clearly state that changes are required in your response but DO NOT append  the APPROVED tag.
    1.  Only append  the APPROVED tag when ALL criteria are satisfied and zero changes are required:
        1. Complete test coverage (≥95%)
        1. No code quality issues, security concerns, or bugs
        1. No bugs or logic errors in the implementation file(s) under test
        1. Full implementation of issue requirements
        1. Compliance with all previous feedback
        1. No unnecessary files

**Note:** Test-focused PRs should maintain clear scope boundaries. However, this does NOT exempt bugs in the specific file(s) being tested - those must be fixed in the same PR. Only bugs in unrelated files may be deferred to separate issues.

Re-evaluate all previously resolved comments to ensure fixes are complete and correct.

**Comment by Cioppolo14** [comment]
@shshaheen Please address code rabbit's [comments](https://github.com/PalisadoesFoundation/talawa/pull/3241#issuecomment-3712970417)

File: test/views/after_auth_screens/chat/widgets/group_member_selector_test.dart [hunk 1]
```diff
@@ -40,778 +42,512 @@ void main() {
...
+      });
 
-      await tester.pumpAndSettle();
+      await pumpSelector(tester, onMembersChanged: (members) {});
 
-      expect(find.text('Group Creator'), findsOneWidget);
+      // Loading indicator should be gone
+      expect(find.byType(CircularProgressIndicator), findsNothing);
+      expect(find.byType(CheckboxListTile), findsAtLeastNWidgets(3));
     });
```
[the remark this answers is not in the document]
Comment (shshaheen) [reply test/views/after_auth_screens/chat/widgets/group_member_selector_test.dart]: This case is already handled in the updated test. Due to refactoring, the diff shows it as a delete/add, so the suggestion seems to be based on the old context rather than the final file.

File: test/views/after_auth_screens/chat/widgets/group_member_selector_test.dart [hunk 2]
```diff
@@ -40,778 +42,498 @@ void main() {
...
```
[the remark this answers is not in the document]
Comment (shshaheen) [reply test/views/after_auth_screens/chat/widgets/group_member_selector_test.dart]: This test focuses on verifying the post-loading UI. The intermediate loading state is skipped due to the use of pumpAndSettle().

**Comment by shshaheen** [comment]
Hi @Cioppolo14 , I’ve addressed all the required CodeRabbit comments. Since the remaining suggestion is optional, I’ve kept the current implementation as is. Please let me know if you think this needs to be addressed as well.

**Comment by palisadoes** [comment]
> Hi @Cioppolo14 , I’ve addressed all the required CodeRabbit comments. Since the remaining suggestion is optional, I’ve kept the current implementation as is. Please let me know if you think this needs to be addressed as well.

Please apply the suggested change. We need to app to be fully tested

**Comment by shshaheen** [comment]
> > Hi @Cioppolo14 , I’ve addressed all the required CodeRabbit comments. Since the remaining suggestion is optional, I’ve kept the current implementation as is. Please let me know if you think this needs to be addressed as well.
> 
> Please apply the suggested change. We need to app to be fully tested

Got it. I’ll apply the suggested change and ensure the loading state is covered in tests.
